### PR TITLE
internationalize controller rescue_from text

### LINKED ISSUE #6534: Internationalize controller rescue_from text
- https://github.com/diaspora/diaspora/blob/develop/app/controllers/posts_controller.rb#L25
- https://github.com/diaspora/diaspora/blob/develop/app/controllers/aspect_memberships_controller.rb#L71
- https://github.com/diaspora/diaspora/blob/develop/app/controllers/aspect_memberships_controller.rb#L75
- https://github.com/diaspora/diaspora/blob/develop/app/controllers/aspect_memberships_controller.rb#L79
- https://github.com/diaspora/diaspora/blob/develop/app/controllers/invitation_codes_controller.rb#L5

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27978652-internationalize-controller-rescue_from-text?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by tarasmatsyk**
This commit moves rescue_from errors in to locales and closes #6534

### CODE REVIEW

File: app/controllers/aspect_memberships_controller.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ def create
   end
 
   rescue_from ActiveRecord::StatementInvalid do
-    render :text => "Duplicate record rejected.", :status => 400
+    render :text => I18n.t('aspect_memberships.destroy.invalid_statement'), :status => 400
```
Comment (diaspora-code-review) [inline app/controllers/aspect_memberships_controller.rb]: Use the new Ruby 1.9 hash syntax.

Comment (diaspora-code-review) [inline app/controllers/aspect_memberships_controller.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: app/controllers/aspect_memberships_controller.rb [hunk 2]
```diff
@@ -76,7 +76,7 @@ def create
   end
 
   rescue_from Diaspora::NotMine do
-    render :text => "You are not allowed to do that.", :status => 403
+    render :text => I18n.t('aspect_memberships.destroy.forbitten'), :status => 403
```
Comment (diaspora-code-review) [inline app/controllers/aspect_memberships_controller.rb]: Use the new Ruby 1.9 hash syntax.

Comment (diaspora-code-review) [inline app/controllers/aspect_memberships_controller.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: app/controllers/invitation_codes_controller.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class InvitationCodesController < ApplicationController
   before_action :ensure_valid_invite_code
 
   rescue_from ActiveRecord::RecordNotFound do
-    redirect_to root_url, :notice => "That invite code is no longer valid"
+    redirect_to root_url, :notice => I18n.t('invintation_codes.not_valid')
```
Comment (diaspora-code-review) [inline app/controllers/invitation_codes_controller.rb]: Use the new Ruby 1.9 hash syntax.

Comment (diaspora-code-review) [inline app/controllers/invitation_codes_controller.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -22,7 +22,7 @@ class PostsController < ApplicationController
   end
 
   rescue_from Diaspora::NotMine do
-    render text: "You are not allowed to do that", status: 403
+    render text: I18n.t('posts.show.forbitten'), status: 403
```
Comment (diaspora-code-review) [inline app/controllers/posts_controller.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 5]
```diff
@@ -1,12 +1,13 @@
-#   Copyright (c) 2010-2011, Diaspora Inc.  This file is
+##   Copyright (c) 2010-2011, Diaspora Inc.  This file is   Copyright (c) 2010-2011, Diaspora Inc.  This file is
 #   licensed under the Affero General Public License version 3 or later.  See
 #   the COPYRIGHT file.
 
 require 'spec_helper'
+require 'pry-byebug'
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 
 describe AspectMembershipsController, :type => :controller do
   before do
-    @aspect0  = alice.aspects.first
+    @aspect0  = alice.aspects.first
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Trailing whitespace detected.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 7]
```diff
@@ -69,6 +70,7 @@
       post :create, params
       post :create, params
       expect(response.status).to eq(400)
+      expect(response.body).to include(I18n.t('aspect_memberships.destroy.invalid_statement'))
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 8]
```diff
@@ -104,7 +106,7 @@
     it 'aspect membership does not exist' do
       delete :destroy, :format => :json, :id => 123
       expect(response).not_to be_success
-      expect(response.body).to include "Could not find the selected person in that aspect"
+      expect(response.body).to include I18n.t('aspect_memberships.destroy.no_membership')
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

File: app/controllers/aspect_memberships_controller.rb (hunk 2 above)
Comment (jhass) [inline app/controllers/aspect_memberships_controller.rb]: `forbidden`

File: app/controllers/invitation_codes_controller.rb (hunk 3 above)
Comment (jhass) [inline app/controllers/invitation_codes_controller.rb]: `invitation`

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 9]
```diff
@@ -1,12 +1,13 @@
-#   Copyright (c) 2010-2011, Diaspora Inc.  This file is
+##   Copyright (c) 2010-2011, Diaspora Inc.  This file is   Copyright (c) 2010-2011, Diaspora Inc.  This file is
```
Comment (jhass) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Typo

File: spec/controllers/aspect_memberships_controller_spec.rb (hunk 5 above)
Comment (jhass) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Unnecessary and unrelated change.

**Comment by tamatsyk** [comment]
@jhass, thank you for comments, I have to be more careful with text editors :+1:

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 10]
```diff
@@ -1,4 +1,4 @@
-#   Copyright (c) 2010-2011, Diaspora Inc.  This file is
+#   Copyright (c) 2010-2011, Diaspora Inc.  This file is   Copyright (c) 2010-2011, Diaspora Inc.  This file is
```
Comment (jhass) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Please revert this change.

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (jhass) [inline app/controllers/posts_controller.rb]: `forbidden`, still.

**Comment by jhass** [comment]
Please also fix the style issues.

**Comment by tamatsyk** [comment]
Ok, I improved a code style of aspect_memberships_controller using rubocop. 
In order to do the same for rspec the test class has to be refactored a bit and I need more time for it.

**Comment by jhass** [comment]
Well, I was talking about  the ones mentioned by @diaspora-code-review only here.

**Comment by tamatsyk** [comment]
I understand, I did not want to mix up different styles like single and double quotes in one file, that's why I updated everything. The same with hash syntax. It is better to have everything consistent. 

Do you want me to roll-back changes? I can send a new pool request.

**Comment by jhass** [comment]
Well, they're on the border of being small enough to let it go through. There are still unaddressed comments though.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 11]
```diff
@@ -15,68 +15,66 @@
...
       @person = eve.person
     end
 
-    it 'succeeds' do
+    it "succeeds" do
       post :create,
-        :format => :json,
-        :person_id => bob.person.id,
-        :aspect_id => @aspect1.id
+        format: json,
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the elements of a hash literal if they span more than one line.

Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 12]
```diff
@@ -15,68 +15,66 @@
...
 
-    it 'creates an aspect membership' do
+    it "creates an aspect membership" do
       expect {
         post :create,
-          :format => :json,
-          :person_id => bob.person.id,
-          :aspect_id => @aspect1.id
-      }.to change{
+          format: json,
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the elements of a hash literal if they span more than one line.

Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 13]
```diff
@@ -15,68 +15,66 @@
...
+    it "creates a contact" do
+      # argggg why?
       alice.contacts.reload
       expect {
         post :create,
-          :format => :json,
-          :person_id => @person.id,
-          :aspect_id => @aspect0.id
-      }.to change{
+          format: json,
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the elements of a hash literal if they span more than one line.

Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 14]
```diff
@@ -15,68 +15,66 @@
...
     end
 
-    it 'failure flashes error' do
+    it "failure flashes error" do
       expect(alice).to receive(:share_with).and_return(nil)
       post :create,
-        :format => :json,
-        :person_id => @person.id,
-        :aspect_id => @aspect0.id
+        format: json,
```
Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the elements of a hash literal if they span more than one line.

Comment (diaspora-code-review) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Align the parameters of a method call if they span more than one line.

File: config/locales/diaspora/en.yml [hunk 15]
```diff
@@ -993,6 +997,7 @@ en:
       destroy: "Delete"
       permalink: "Permalink"
       not_found: "Sorry, we couldn’t find that post."
+      forbitten: "You are not allowed to do that"
```
Comment (jhass) [inline config/locales/diaspora/en.yml]: `forbidden`.

Remember to fix the new style issues too.

Comment (tamatsyk) [inline config/locales/diaspora/en.yml]: Please excuse me for taking too much your time. 
Hope everything is fine now.

File: spec/controllers/aspect_memberships_controller_spec.rb [hunk 16]
```diff
@@ -2,12 +2,12 @@
 #   licensed under the Affero General Public License version 3 or later.  See
 #   the COPYRIGHT file.
 
-require 'spec_helper'
+require "spec_helper"
 
-describe AspectMembershipsController, :type => :controller do
+describe AspectMembershipsController, type: controller do
```
Comment (jhass) [inline spec/controllers/aspect_memberships_controller_spec.rb]: Missing `:`. Have a look at our wiki on how to run the tests locally: https://wiki.diasporafoundation.org/Testing_workflow

**Comment by svbergerem** [comment]
@tarasmatsyk Could you please squash your commits?

**Comment by tarasmatsyk** [comment]
@svbergerem, oh, thank you very much for following up. I thought I over engineered this one.
And I did not know there is such thing as squash. Thanks for helping me to learn.
Did I make everything correctly?

**Comment by svbergerem** [comment]
@tarasmatsyk Your code is looking good but you didn't squash your commits into one which would be an appropriate number of commits for your changes. Here is how you can squash the commits:

```bash
git remote add upstream https://github.com/diaspora/diaspora.git
git fetch upstream
git checkout issue-6534
git rebase -i upstream/develop
# Leave the first commit at pick, choose squash or fix for the others.
# If there are any conflicts git will inform you about that, resolve them for
# example with git mergetool or manually and with git add, then run
# git rebase --continue, again  watch the output since that might happen for
# each commit.
git log # Look at the updated commit history
git status # should report you're back on your branch
# We need to force push since a rebase edits the commits, effectively
# creating new ones. This is okay to do in a feature branch, the pull
# request will update.
git push -f origin issue-6534 # Assuming you cloned from your fork
```

There was no review for about two months because no one got notified that you fixed the remaining issues in your code. Please add a comment when you push some changes and you think that your code is ready for another review or you need some help. In that case we will receive a notification and have another look at your code.

**Comment by tarasmatsyk** [comment]
@svbergerem, good evening,
looks like I got it but I am not sure if it was done correctly. 
I would appreciate any feedback.

**Comment by svbergerem** [comment]
Merged as 960e651764f08f342225f31ed8cdd79a9c9bd53b.

Thank you! :)

**Comment by tarasmatsyk** [comment]
Yeeeey!! My first commit to open source))
@svbergerem, @jhass, thank you for your patience!

**Comment by theworldbright** [comment]
@tarasmatsyk Congratz :+1:

**Comment by Flaburgan** [comment]
Nice work @tarasmatsyk congratulation!

**Comment by SansPseudoFix** [comment]
> Yeeeey!! My first commit to open source))

So, welcome and well done @tarasmatsyk \o/

**Comment by dadall** [comment]
@tarasmatsyk : Wow, congrats !
